docs/testplan/SmartSwitch-PMON-testplan.md

```diff
@@ -0,0 +1,939 @@
+ * get_num_modules(self):
+    * Should return number of DPUs + 1 switch
+ * get_module(self, index):
+    * Make sure for each index this API returns an object and has some content and not None
+ * get_all_modules(self):
+    * This should return a list of items
+ * get_module_index(self, module_name):
+    * Given the module name say “DPU0” should return the index of it “1”
+ 
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+    is_modular_chassis(self):
+    Output: False
+
+    get_num_modules(self):
+    Output: number of DPUs + 1
+
+    get_module(self, DPU0):
+    Output: DPU0 object
+
+    get_all_modules(self):
+    Output: list of objects (one per DPU + 1 switch object)
+
+    get_module_index(self, DPU0):
+    Output: could be any value from 0 to modules count -1
+```
+#### Pass/Fail Criteria
+ * The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output"
+
+### 1.3 Check DpuModule APIs for SmartSwitch
+
+#### Steps
+ * get_dpu_id(self):
+    * Should return ID of the DpuModule Ex: 1 on DPU0
+ * get_reboot_cause(self):
+    * Reboot the module and then execute the "show reboot-cause ..." CLIs
+    * Verify the output string shows the correct Time and Cause
+    * Limit the testing to software reboot
+ * get_state_info(self):
+    * This should return an object
+    * Stop one of the DPU containers on this DPU
+    * Execute the CLI and check the dpu-controlplane value should be down
+ * get_health_info(self):
+    * This should return an object
+    * Stop one of the DPU containers on this DPU
+    * Execute the CLI and check if the health shows the stopped container
+ 
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+    get_dpu_id(self):
+    Output: When on module DPUx should return x+1
+
+    get_reboot_cause(self):
+    Output: {"Device": "DPU0", "Name": 2024_05_31_00_33_30, "Cause":  "reboot", "Time": "Fri 31 May 2024 12:29:34 AM UTC", "User": "NA", "Comment": "NA"}
+
+    get_state_info(self):
+    Output: dpu state info object
+
+    get_health_info(self):
+    Output: dpu health info object
+
+```
+#### Pass/Fail Criteria
+ * Verify that all the APIs mentioned return the expected output
+
+### 1.4 Check modified ModuleClass APIs
+
+#### Steps
+ * get_base_mac(self):
+    * Should return the base mac address of this DPU
+    * Read all DPUs mac and verify if they are unique and not None
+ * get_system_eeprom_info(self):
+    * Verify the returned dictionary key:value
+ * get_name(self):
+    * Verify if this API returns “DPU0” to “DPU7” on each of them
+ * get_description(self):
+    * Should return a string
+ * get_type(self):
+    * Should return “DPU” which is “MODULE_TYPE_DPU”
+ * get_oper_status(self):
+    * Should return the operational status of the DPU
+    * Stop one ore more containers
+    * Execute the CLI and see if it is down
+ * reboot(self, reboot_type):
+    * Issue this CLI with input “MODULE_REBOOT_DEFAULT”
```

Comment:
The reboot type should be updated based on SS reboot HLD https://github.com/sonic-net/SONiC/pull/1699

Reply:
Agree, added this check.